Makefile subfolders

plmorange opened the pull request:
<!--
The RIOT community cares a lot about code quality.
Therefore, before describing what your contribution is about, we would like
you to make sure that your modifications are compliant with the RIOT
coding conventions, see https://github.com/RIOT-OS/RIOT/blob/master/CODING_CONVENTIONS.md.
-->

### Contribution description

<!--
Put here the description of your contribution:
- describe which part(s) of RIOT is (are) involved
- if it's a bug fix, describe the bug that it solves and how it is solved
- you can also give more information to reviewers about how to test your changes
-->


### Testing procedure

<!--
Details steps to test your contribution:
- which test/example to compile for which board and is there a 'test' command
- how to know that it was not working/available in master
- the expected success test output
-->


### Issues/PRs references

<!--
Examples: Fixes #1234. See also #5678. Depends on PR #9876.

Please use keywords (e.g., fixes, resolve) with the links to the issues you
resolved, this way they will be automatically closed when your pull request
is merged. See https://help.github.com/articles/closing-issues-using-keywords/.
-->

mguetschow committed:
Makefiles: (application) support for SRC in subfolders

mguetschow committed:
Makefiles: (application) support for SRC in subfolders

mguetschow committed:
add example showcasing application subfolders

mguetschow committed:
add example showcasing application subfolders

plmorange committed:
Merge branch 'mguetschow:makefile-subfolders' into makefile-subfolders

plmorange committed:
Add tests for SRC in makefile

mguetschow commented:
Hej, thanks for your work on this!

Your test application doubles some part from the `examples/subfolders` application, I think it would make sense to combine both into a single test application? I would now rather put it in `tests/build_system` as you did instead of `examples`. In any case, the test application needs a README explaining its purpose and structure.

Would you mind combining them into one? Also, it might be nice to find a suitable point in the documentation to add information to about the `SRC` variable, maybe in https://doc.riot-os.org/creating-an-application.html or https://doc.riot-os.org/build-system-basics.html.

plmorange committed:
Remove duplicate example

plmorange committed:
Add usage of SRC in doc

plmorange commented:
Thanks for your feedback !

I saw the feedback of maribu (https://github.com/RIOT-OS/RIOT/pull/20024#issuecomment-2452501865) so I've removed all duplicates between test and example and added doc in `create-an-application.md`

mguetschow commented:
Thanks, I'm gonna merge this into my branch and then rework it a bit. I think the documentation part could be shorter by simply referring to the example's README.
